frontend/Interfaces/buddy/DIP/imgcodecs/Replenishment.hpp

```diff
@@ -0,0 +1,181 @@
+//
+// Copyright (C) 2000, Intel Corporation, all rights reserved.
+// Third party copyrights are property of their respective owners.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+//   * Redistribution's of source code must retain the above copyright notice,
+//     this list of conditions and the following disclaimer.
+//
+//   * Redistribution's in binary form must reproduce the above copyright
+//   notice,
+//     this list of conditions and the following disclaimer in the documentation
+//     and/or other materials provided with the distribution.
+//
+//   * The name of Intel Corporation may not be used to endorse or promote
+//   products
+//     derived from this software without specific prior written permission.
+//
+// This software is provided by the copyright holders and contributors "as is"
+// and any express or implied warranties, including, but not limited to, the
+// implied warranties of merchantability and fitness for a particular purpose
+// are disclaimed. In no event shall the Intel Corporation or contributors be
+// liable for any direct, indirect, incidental, special, exemplary, or
+// consequential damages (including, but not limited to, procurement of
+// substitute goods or services; loss of use, data, or profits; or business
+// interruption) however caused and on any theory of liability, whether in
+// contract, strict liability, or tort (including negligence or otherwise)
+// arising in any way out of the use of this software, even if advised of the
+// possibility of such damage.
+//
+//===----------------------------------------------------------------------===//
+//
+// This file is modified from opencv's
+// modules/imgcodecs/include/opencv2/imgcodecs.hpp file
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _REPLEISHMENT_H
+#define _REPLEISHMENT_H
+
+#include <cassert>
+#include <memory>
+#include <string>
+#include <type_traits>
+
+using namespace std;
+typedef unsigned long ulong;
+typedef unsigned char uchar;
+typedef unsigned short ushort;
+typedef unsigned int uint;
+typedef string String;
+
+namespace dip {
+#define IMG_CN_MAX 512
+#define IMG_CN_SHIFT 3
+#define IMG_DEPTH_MAX (1 << IMG_CN_SHIFT)
+#define IMG_MAT_CN_MASK ((IMG_CN_MAX - 1) << IMG_CN_SHIFT)
+#define IMG_MAT_CN(_type) ((((_type)&IMG_MAT_CN_MASK) >> IMG_CN_SHIFT) + 1)
+
+#define IMG_8U 0
+#define IMG_8S 1
+#define IMG_16U 2
+#define IMG_16S 3
+#define IMG_32S 4
+#define IMG_32F 5
+#define IMG_64F 6
+#define IMG_16F 7
+
+#define IMG_MAT_DEPTH_MASK (IMG_DEPTH_MAX - 1)
+#define IMG_MAT_DEPTH(flags) ((flags)&IMG_MAT_DEPTH_MASK)
+
+#define IMG_MAKETYPE(depth, cn)                                                \
+  (IMG_MAT_DEPTH(depth) + (((cn)-1) << IMG_CN_SHIFT))
+#define IMG_MAKE_TYPE IMG_MAKETYPE
+
+#define IMG_8UC1 IMG_MAKETYPE(IMG_8U, 1)
```

Comment:
Do we need these types? Our image container template class can already represent such information through types and dimensions.

Reply:
Can we get types dynamically from template classes?

Reply:
Currently, these types are used in two places during bmp image parsing
1.bmp decode, in the readHeader() function: this->m_type =
iscolor ?  ((m_bpp == 32 && m_rle_code ! = BMP_RGB) ?  IMG_8UC4 : IMG_8UC3)
: IMG_8UC1;
2. The imread function modifies the channel according to flag and the type obtained above to determine the storage space of image data.
If we remove these types, we need to calculate channels in a different way.